src/Validator/PathValidator.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli\Validator;
+
+use InvalidArgumentException;
+use RuntimeException;
+
+use function file_exists;
+use function gettype;
+use function in_array;
+use function is_dir;
+use function is_string;
+use function sprintf;
+
+/**
+ * @internal
+ */
+final class PathValidator
+{
+    public const TYPE_DIR = 'dir';
+    public const TYPE_FILE = 'file';
```

Comment:
Add another, `public const TYPE_ANY = null;` or similar, so people can use a constant to indicate when they allow either files or directories.